jina/serve/executors/__init__.py

```diff
@@ -152,17 +155,19 @@ def __init__(
         if type(self) == BaseExecutor:
             self.requests[__default_endpoint__] = self._dry_run_func
 
+        self._lock = asyncio.Lock()  # Lock to run in Executor non async methods in a way that does not block the event loop to do health checks without the fear of having race conditions or multithreading issues.
```

Comment:
I'm not sure if this is solving the health check issue. The health check methods/stubs are implemented by the WorkerRuntime so I'm not sure how the request would be forwarded to the Executor.

Reply:
this unblocks the `event loop` so that the Health Check can be processed

Reply:
this is not solving it, this solves potential threading issues coming from the fact that we have `run_in_executor`